spark/spark-2.1/src/main/scala/org/apache/spark/sql/hive/HivemallOps.scala

```diff
@@ -805,6 +805,47 @@ final class HivemallOps(df: DataFrame) extends Logging {
     JoinTopK(kInt, df.logicalPlan, right.logicalPlan, Inner, Option(joinExprs.expr))(score.named)
   }
 
+  private def doFlatten(schema: StructType, prefix: Option[String] = None) : Seq[Column] = {
+    schema.fields.flatMap { f =>
+      val colName = prefix.map(p => s"$p.${f.name}").getOrElse(f.name)
```

Comment:
`ds.select($"_2._2")` works for nested scheme. So, flatten'ed `ds.flatten` should be accessed without `.`.